src/main/java/org/cyclonedx/util/serializer/LicenseChoiceSerializer.java

```diff
@@ -28,48 +28,108 @@
+    this(LicenseChoice.class, isXml);
+    this.isXml = isXml;
   }
 
-  public LicenseChoiceSerializer(final Class<LicenseChoice> t) {
+  public LicenseChoiceSerializer(Class<LicenseChoice> t, boolean isXml) {
     super(t);
+    this.isXml = isXml;
   }
 
   @Override
   public void serialize(
-      final LicenseChoice licenseChoice, final JsonGenerator gen, final SerializerProvider provider)
+      final LicenseChoice lc, final JsonGenerator gen, final SerializerProvider provider)
       throws IOException
   {
-    if (licenseChoice == null) {
+    if (lc == null) {
       return;
     }
 
-    if (gen instanceof ToXmlGenerator) {
-      serializeXml(licenseChoice, gen, provider);
+    if (isXml && gen instanceof ToXmlGenerator) {
+      ToXmlGenerator toXmlGenerator = (ToXmlGenerator) gen;
+      serializeXml(toXmlGenerator, lc);
     }
     else {
-      serializeJson(licenseChoice, gen, provider);
+      serializeJson(lc, gen, provider);
     }
   }
 
-  private void serializeXml(
-      final LicenseChoice licenseChoice, final JsonGenerator gen, final SerializerProvider provider)
-      throws IOException
-  {
-      if (CollectionUtils.isNotEmpty(licenseChoice.getLicenses())) {
-        serializeLicensesToJsonArray(licenseChoice, gen, provider);
-      }
-      else if (licenseChoice.getExpression() != null &&
-          StringUtils.isNotBlank(licenseChoice.getExpression().getValue())) {
-        final ToXmlGenerator toXmlGenerator = (ToXmlGenerator) gen;
-        serializeExpressionToXml(licenseChoice, toXmlGenerator);
+    private void serializeXml(ToXmlGenerator toXmlGenerator, LicenseChoice lc) throws IOException {
+    if (CollectionUtils.isNotEmpty(lc.getLicenses())) {
+      toXmlGenerator.writeStartObject();
+      toXmlGenerator.writeFieldName("license");
+      toXmlGenerator.writeStartArray();
+      for (License l : lc.getLicenses()) {
+        serializeXmlAttributes(toXmlGenerator, l.getBomRef(), l.getAcknowledgement());
+
+        if (StringUtils.isNotBlank(l.getId())) {
+          toXmlGenerator.writeStringField("id", l.getId());
+        }
+        else if (StringUtils.isNotBlank(l.getName())) {
+          toXmlGenerator.writeStringField("name", l.getName());
+        }
+
+        if (l.getLicensing() != null) {
+          toXmlGenerator.writeObjectField("licensing", l.getLicensing());
+        }
+
+        if (l.getAttachmentText() != null) {
+          toXmlGenerator.writeObjectField("text", l.getAttachmentText());
+        }
+
+        if (StringUtils.isNotBlank(l.getUrl())) {
+          toXmlGenerator.writeStringField("url", l.getUrl());
+        }
+
+        if (CollectionUtils.isNotEmpty(l.getProperties())) {
+          toXmlGenerator.writeFieldName("properties");
+          toXmlGenerator.writeStartObject();
+
+          for (Property property : l.getProperties()) {
+            toXmlGenerator.writeObjectField("property", property);
+          }
+          toXmlGenerator.writeEndObject();
+        }
```

Comment:
I think we need to account for older spec versions here.

`acknowledgement` is new as of v1.6, `licensing` and `properties` is new as of v1.5, and so on.

For example, changing `BomXmlGeneratorTest#testIssue408Regression` to generate and validate for v1.5 fails with:

```
org.cyclonedx.exception.ParseException: cvc-complex-type.3.2.2: Attribute 'acknowledgement' is not allowed to appear in element 'license'.
org.cyclonedx.exception.ParseException: cvc-complex-type.3.2.2: Attribute 'acknowledgement' is not allowed to appear in element 'expression'.
```

Reply:
Thanks @mr-zepol! 

But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.

Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

Reply:
> Thanks @mr-zepol!
> 
> But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.
> 
> Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

I get it, you are right, thinking about it I think the filter for each version is not applied for the serializer, I will check how can I do this at least for the license and later for the other ones, thanks

Reply:
> Thanks @mr-zepol!
> 
> But shouldn't parsing a v1.6 BOM, and generating a v1.5 BOM based on that not yield a v1.5-compatible BOM? The tests currently assert that parsing a v1.5 yields a valid v1.5 BOM, and parsing a v1.6 yields a valid v1.6 BOM.
> 
> Parsing `issue408.xml` (v1.6) and generating a v1.5 BOM from that gives a `license` node with `acknowledgment` attribute, which fails v1.5 validation.

Added the validations d4d561cad1a6db52431522490a3ece603612ac7a , and also added tests from 1.6 to 1.4 for both XML and JSON